File: src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionFormParseTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;
public class TransformerFunctionFormParseTest extends BaseTest {
    @Test
    void inline() {
        assertTransformation("a=1&b=B&c", "$$formparse:$", fromJson("""
            {
              "a": "1",
              "a$$": ["1"],
              "b": "B",
              "b$$": ["B"],
              "c": "true",
              "c$$": ["true"]
            }"""));

        // arrays
        assertTransformation("a=1&a=2", "$$formparse:$", fromJson("""
            {
              "a": "1",
              "a$$": ["1","2"]
            }"""));
    }

    @Test
    void object() {
        assertTransformation("a=1&b=B&c", fromJson("""
        { "$$formparse": "$" }"""), fromJson("""
            {
              "a": "1",
              "a$$": ["1"],
              "b": "B",
              "b$$": ["B"],
              "c": "true",
              "c$$": ["true"]
            }"""));

        // arrays
        assertTransformation("a=1&a=2", fromJson("""
        { "$$formparse": "$" }"""), fromJson("""
            {
              "a": "1",
              "a$$": ["1","2"]
            }"""));
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionFormParse.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionCsv", () => {
  test("inline", async () => {
    await assertTransformation("a=1&b=B&c", "$$formparse:$", {
      a: "1",
      a$$: ["1"],
      b: "B",
      b$$: ["B"],
      c: "true",
      c$$: ["true"],
    });

    // arrays
    await assertTransformation("a=1&a=2", "$$formparse:$", {
      a: "1",
      a$$: ["1", "2"],
    });
  });

  test("object", async () => {
    await assertTransformation(
      "a=1&b=B&c",
      { $$formparse: "$" },
      {
        a: "1",
        a$$: ["1"],
        b: "B",
        b$$: ["B"],
        c: "true",
        c$$: ["true"],
      },
    );

    // arrays
    await assertTransformation(
      "a=1&a=2",
      { $$formparse: "$" },
      {
        a: "1",
        a$$: ["1", "2"],
      },
    );
  });
});
